### PR TITLE
Added Rust to CI + libsodium-neon logging

### PR DESCRIPTION
**Pull request by ffflorian**

### CODE REVIEW

File: electron/js/preload.js [hunk 1]
```diff
@@ -139,8 +139,10 @@ ipcRenderer.once('webapp-loaded', function(sender, config) {
   // else we are on /auth
   try {
     Object.assign(window.sodium, require('libsodium-neon'));
+    console.log('Using libsodium-neon.');
   } catch (err) {
     // fallback to libsodium.js
+    console.log('Failed loading libsodium-neon', err);
```
Comment (lipis) [inline electron/js/preload.js]: `console.error(...)`?

File: electron/js/preload.js [hunk 2]
```diff
@@ -139,8 +139,10 @@ ipcRenderer.once('webapp-loaded', function(sender, config) {
   // else we are on /auth
   try {
     Object.assign(window.sodium, require('libsodium-neon'));
+    console.log('Using libsodium-neon.');
```
Comment (lipis) [inline electron/js/preload.js]: Remove the `.` (or add it the error)

Comment (lipis) [inline electron/js/preload.js]: On a second thought.. we don't use dots anywhere as far as I know.. so let's just remove it

File: electron/js/preload.js [hunk 3]
```diff
@@ -139,8 +139,10 @@ ipcRenderer.once('webapp-loaded', function(sender, config) {
   // else we are on /auth
   try {
     Object.assign(window.sodium, require('libsodium-neon'));
+    console.log('Using libsodium-neon');
   } catch (err) {
     // fallback to libsodium.js
+    console.error('Failed loading libsodium-neon', err);
```
Comment (bennyn) [inline electron/js/preload.js]: You can make use of the error message:
```js
console.error(`Failed loading libsodium-neon: ${err.message}`, err);
```
The vairable `err` should be also named `error`.

File: electron/js/preload.js (hunk 1 above)
Comment (bennyn) [inline electron/js/preload.js]: @lipis Will `console.error` be written into our `console.log` debug file?

**Pull request closed without merging**